### PR TITLE
test: dualVMToken uniswap test

### PR DESCRIPTION
**Pull request by obatirou**
Time spent on this PR:

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): testing

## What is the current behavior?

Resolves #<Issue number>

## What is the new behavior?

- test events sent from DualVMToken
- test uniswapV2 with DualVMToken

- - -
This change is [Reviewable](https://reviewable.io/reviews/kkrt-labs/kakarot/1415)

### CODE REVIEW

File: tests/end_to_end/conftest.py [hunk 1]
```diff
@@ -195,3 +196,48 @@ def relayers(worker_id):
     except ValueError:
         logger.info(f"Error while setting relayer index to {worker_id}")
     return
+
+
+# Uniswap fixtures
```
Comment (obatirou) [inline tests/end_to_end/conftest.py:201]: @ClementWalter would like your view on this: do you find this ok to import uniswap fixtures in the end_to_end conftest and delete the uniswap conftest ?
It is necessary as a test with DualTokenVM is using UniswapRouter to ensure adding liquidity and swap works.

File: tests/end_to_end/conftest.py [hunk 2]
```diff
@@ -195,3 +196,48 @@ def relayers(worker_id):
...
+
+TOTAL_SUPPLY = 10000 * 10**18
+
+
+@pytest_asyncio.fixture(scope="function")
```
Comment (ClementWalter) [inline tests/end_to_end/conftest.py:206]: do you really want to deploy a new token for each test?

File: tests/end_to_end/conftest.py [hunk 3]
```diff
@@ -195,3 +196,48 @@ def relayers(worker_id):
...
+
+
+@pytest_asyncio.fixture(scope="module")
+async def router(owner, factory, weth):
+    return await deploy_kakarot(
+        "UniswapV2Router",
+        "UniswapV2Router02",
+        factory.address,
+        weth.address,
+        caller_eoa=owner.starknet_contract,
```
Comment (ClementWalter) [inline tests/end_to_end/conftest.py:242]: for all these tokens, it `msg.sender` is not used in the constructor, remove the `caller_eao` kwarg to make it clear that it's doesn't matter

File: tests/end_to_end/conftest.py (hunk 2 above)
Comment (obatirou) [reply tests/end_to_end/conftest.py:206]: Yes this is necessary for the UniswapV2 ERC20 tests

**Pull request merged**